### DISCUSSION TITLE
Storing configuration options inside todo.txt file

### DISCUSSION THREAD

**Discussion by xuhcc** [todotxt/todo.txt#50] [Q&A]
I'm looking for a way to store some configuration options in todo.txt file itself.

Several apps already [have support](https://docs.google.com/spreadsheets/d/11nqV2j8qc_O5SVsvkZ2JGZP98CABLQ6EwBxAQQi7Huc/edit#gid=0) for the `h` tag, so I think that we can use hidden tasks to define configuration options while maintaining compatibility with the current todo.txt v1 specification. For example, here's how options could be defined using the `config` tag:

```js
// With JSON-like option format
h:1 config:{"title":"myTodoFile"}
h:1 config:{"colored-projects":{"project1":"#eee","project2":"#ccc"}}

// Or INI-like option format
h:1 config:title=myTodoFile
h:1 config:colored-projects=project1,#eee;project2,#ccc
```

Is there any todo.txt app that follows this approach?

**Comment by pablito1755** [comment]
Why not simply define a new standalone standard configuration file which would live in the same folder as todo.txt, done.txt, etc.? I think this may have been mentioned in one of the other issue threads.

**Comment by xuhcc** [comment]
Yes, standalone config file is a good approach (for example [todo.txt CLI](https://github.com/todotxt/todo.txt-cli/blob/master/todo.cfg) already has its own config format). But I would prefer to put everything in a single file as it is more appropriate for the mobile app I'm building.

**Comment by TangentFoxy** [comment]
@xuhcc The doc you linked to simply calls `h` configurable. What is the purpose of the `1` in your example? Is it just "a value" to be used with `key:value` style tags?

**Comment by xuhcc** [comment]
@TangentFoxy Yes, it's a value. For example, [simpletask](https://github.com/mpcjanssen/simpletask-android/blob/master/app/src/main/assets/index.en.md#extensions) supports this notation.
